File: mobile/src/services/sign.ts
import api from "./api";
import type { SpeakResponse } from "../types/api";

export async function triggerSOS(location?: {
  latitude: number;
  longitude: number;
}): Promise<SpeakResponse> {
  const response = await api.post<SpeakResponse>("/sign/sos", location ?? {});
  return response.data;
}

export async function manualSpeak(
  text: string,
  emotion: string = "neutral",
  language: string = "en"
): Promise<SpeakResponse> {
  const response = await api.post<SpeakResponse>("/sign/speak", {
    text,
    emotion,
    language,
  });
  return response.data;
}

File: backend/app/schemas/sign.py
from pydantic import BaseModel


class SpeakRequest(BaseModel):
    text: str
    emotion: str = "neutral"
    language: str = "en"


class SpeakResponse(BaseModel):
    sentence: str
    audio_url: str


class SOSRequest(BaseModel):
    latitude: float | None = None
    longitude: float | None = None


class TranscriptEntryResponse(BaseModel):
    id: str
    text: str
    raw_letters: str | None
    emotion: str
    language: str
    created_at: str
